encoders/src/test/java/au/csiro/pathling/encoders/LightweightFhirEncodersTest.java

```diff
@@ -205,4 +208,55 @@ public void testEncodesExtensions() {
     assertNotNull(stageTypeExtensions);
     assertStringExtension("uuid:ext12", "ext12", (Row) stageTypeExtensions.apply(0));
   }
+
+  @Test
+  public void testQuantityCanonicalization() {
+    final ExpressionEncoder<Observation> encoder = fhirEncoders.of(Observation.class);
+    final Observation observation = TestData.newUcumObservation();
+
+    final ExpressionEncoder<Observation> resolvedEncoder = EncoderUtils.defaultResolveAndBind(
+        encoder);
+    final InternalRow serializedRow = resolvedEncoder.createSerializer().apply(observation);
+
+    final ExpressionEncoder<Row> rowEncoder = EncoderUtils.defaultResolveAndBind(
+        RowEncoder.apply(encoder.schema()));
+    final Row observationRow = rowEncoder.createDeserializer().apply(serializedRow);
+
+    final Row quantityRow = observationRow.getStruct(observationRow.fieldIndex("valueQuantity"));
+    assertQuantity(quantityRow, "76000.000000", "g");
+  }
```

Comment:
Maybe it would be better to test agains the original quantity making sure that both the canocalized and original fields are encoded/decoded correctly.

Reply:
This is actually canonicalisation test, so the assertion is appropriate.